src/Microsoft.ML.StandardLearners/FactorizationMachine/FactorizationMachineInterface.cs

```diff
@@ -31,12 +31,12 @@ private static bool Compat(AlignedArray a)
         }
 
         [DllImport(NativePath), SuppressUnmanagedCodeSecurity]
-        public static extern void CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, int* /*const*/ fieldIndices, int* /*const*/ featureIndices,
-            float* /*const*/ featureValues, float* /*const*/ linearWeights, float* /*const*/ latentWeights, float* latentSum, float* response);
+        public static extern void CalculateIntermediateVariablesNative(int fieldCount, int latentDim, int count, int* fieldIndices, int* featureIndices,
+            float* featureValues, float* linearWeights, float* latentWeights, float* latentSum, float* response);
 
         [DllImport(NativePath), SuppressUnmanagedCodeSecurity]
         public static extern void CalculateGradientAndUpdateNative(float lambdaLinear, float lambdaLatent, float learningRate, int fieldCount, int latentDim, float weight,
-            int count, int* /*const*/ fieldIndices, int* /*const*/ featureIndices, float* /*const*/ featureValues, float* /*const*/ latentSum, float slope,
+            int count, int* fieldIndices, int* featureIndices, float* featureValues, float* latentSum, float slope,
```

Comment:
[https://github.com/dotnet/machinelearning/pull/383#discussion_r196948210](https://github.com/dotnet/machinelearning/pull/383#discussion_r196948210)
See @wschin comment.
I agree this is not the best solution, but I would prefer to preserve this information until we figure out better one.
Same states for Thunk.cs.

We have them for a reason.